### ISSUE TITLE
Improve Gateway link handling

### ISSUE THREAD

**Issue by code-asher** [coder/jetbrains-coder#292]
- [ ] Start the workspace if it is off
- [ ] Wait for the agent if already starting
- [ ] Add more fallbacks for missing parameters
- [ ] Validate the remote path provided

**Comment by MoeRT09** [comment]
Would it be possible to open a folder selector if the "folder" parameter is omitted? This would make it much easier to select a different folder then the home directory as the project root.

**Comment by code-asher** [comment]
Yup, I would like to add that as well.

**Comment by code-asher** [comment]
I have no idea, can `coder ssh` take agent IDs?  I would imagine 
the names are more human-readable anyway.  Or if we get an agent 
ID we can probably look it up in the plugin and translate to an 
agent name.

Is there a specific use case you trying to solve by passing 
Gateway an ID instead of a name?  Does the template only give you 
access to the ID or something like that?

**Comment by matifali** [comment]
@code-asher See these here for example, We need both the agent_id and name to create the coder_app

https://github.com/coder/modules/blob/abb51a37e1564ea554144beb4fd9d5fb22f28085/jetbrains-gateway/main.tf#L126

I was hoping we could get the agent name from the agent ID somehow.

**Comment by code-asher** [comment]
Oh, I see.  Yeah we can make it take either an ID or a name.  I will add it to the list!

**Comment by matifali** [comment]
@code-asher @mafredri also mentioned [here](https://github.com/coder/coder/pull/9890#discussion_r1340098981) why we need both the id and name.
Our current implementation of the [jetbrains_gateway module](https://registry.coder.com/modules/jetbrains-gateway) also uses both the name and the id. I think we should remove the name.

**Comment by code-asher** [comment]
Yeah it should be easy enough, we already query all the agents so we can just get the name by ID from that API response.
